src/runtime/worker_handle.rs

```diff
@@ -18,24 +18,31 @@ impl WorkerHandle {
 /// Listener used to determine when all handles have been dropped.
 #[derive(Debug)]
 pub(crate) struct WorkerHandleListener {
-    receiver: mpsc::Receiver<()>,
+    sender: watch::Sender<()>,
```

Comment:
This was rewritten to use a `watch` channel which is more appropriate for this type and also allowed us to use `&self` everywhere instead of `&mut self`. The functionality is otherwise unchanged.

Reply:
ahh that all makes sense. thanks!